embrace-gradle-plugin/src/main/java/io/embrace/android/gradle/plugin/tasks/ndk/CompressAndHashSharedObjectFilesTaskRegistration.kt

```diff
@@ -0,0 +1,110 @@
+package io.embrace.android.gradle.plugin.tasks.ndk
+
+import com.android.build.api.variant.Variant
+import io.embrace.android.gradle.plugin.config.PluginBehavior
+import io.embrace.android.gradle.plugin.config.ProjectType
+import io.embrace.android.gradle.plugin.config.UnitySymbolsDir
+import io.embrace.android.gradle.plugin.gradle.registerTask
+import io.embrace.android.gradle.plugin.model.AndroidCompactedVariantData
+import io.embrace.android.gradle.plugin.tasks.il2cpp.UnitySymbolFilesManager
+import io.embrace.android.gradle.plugin.tasks.registration.EmbraceTaskRegistration
+import io.embrace.android.gradle.plugin.tasks.registration.RegistrationParams
+import io.embrace.android.gradle.plugin.util.capitalizedString
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+import java.io.File
+
+class CompressAndHashSharedObjectFilesTaskRegistration(
+    private val behavior: PluginBehavior,
+    private val unitySymbolsDir: Provider<UnitySymbolsDir>,
+    private val projectType: Provider<ProjectType>,
+) : EmbraceTaskRegistration {
+    override fun register(params: RegistrationParams) {
+        params.execute()
+    }
+
+    fun RegistrationParams.execute() {
+        val variantConfig = variantConfigurationsListProperty.get().first { it.variantName == variant.name }
+
+        // Skip registration if NDK is disabled
+        if (variantConfig.embraceConfig?.ndkEnabled == false) return
+
+        val sharedObjectFilesProvider = getSharedObjectFilesProvider(project, data, variant)
+
+        project.registerTask(
+            CompressAndHashSharedObjectFilesTask.NAME,
+            CompressAndHashSharedObjectFilesTask::class.java,
+            data
+        ) { task ->
+            task.onlyIf { projectType.orNull == ProjectType.NATIVE || projectType.orNull == ProjectType.UNITY }
+
+            task.architecturesDirectory.set(project.layout.dir(sharedObjectFilesProvider))
+        }
+    }
+
+    private fun getSharedObjectFilesProvider(
+        project: Project,
+        data: AndroidCompactedVariantData,
+        variant: Variant
+    ): Provider<File> {
+        return projectType.map {
+            when (it) {
+                ProjectType.UNITY -> getUnitySharedObjectFiles(project, data)
+                ProjectType.NATIVE -> getNativeSharedObjectFiles(project, variant)
+                else -> error("Project type must be either UNITY or NATIVE")
+            }
+        }
+    }
+
+    private fun getUnitySharedObjectFiles(project: Project, data: AndroidCompactedVariantData): File {
+        // TODO: Verify if errors should be thrown if the directories or SO files are not found. Improve error messaging.
```

Comment:
Are these todo comments new or were they present in the previous code that's been adapted into this source file?

Reply:
these are new, I'm adding them so I can then think about any uncommon codepaths